### DISCUSSION TITLE
What is the difference between rehype-react and react-markdown?

### DISCUSSION THREAD

**Discussion by tchak** [unifiedjs/unified#171] [Q&A] [answered]
From the README on both projects, it seems like they do exactly the same thing. At first I thought that react-markdown is using rehype-react under the hood but it seems that it is not the case. Is one of the projects deprecated? Which one should be used going forward? react-markdown seems to have more activity. If both projects meant to have a future in parallel, I would love a paragraph in both README explaining why there are two projects :)

**Comment by wooorm** [comment, marked as answer]
Hi Paul!

We’re currently reworking all docs to include information on such things: What is this? When should I use this? Takes a lot of work though, currently in remark.

`react-markdown` was already based on remark, but it wasn’t actively maintained. We started maintaining it about a year or so ago, and hence the activity: cleaning many things and upgrading everything to fit better with the rest of the ecosystem.
It doesn’t yet use `rehype-react` (which is a single plugin), It could potentially in the future, but `react-markdown` is focussed on newcomers that don’t know about unified/remark yet, and has many more features for them.

For a modern replacement of `react-markdown`, you might want to try https://github.com/remarkjs/react-remark instead.
But, for more complex pipelines and with more knowledge of unified, you can do things manually and use `rehype-react`!
